index.html

```diff
@@ -1,390 +1,1471 @@
+      <li>
+        Let |pullAlgorithm| be the following steps:
+        <ol>
+          <li>
+            Let |desiredSize| be the
+            <a href="https://streams.spec.whatwg.org/#desired-size-to-fill-a-streams-internal-queue">desired size</a>
+            of [=this=].{{[[readable]]}}'s
+            <a href="https://streams.spec.whatwg.org/#internal-queues">internal queue</a>.
+          <li>
+            Run the following steps [=in parallel=]:
+            <ol>
+              <li>
+                Invoke the operating system to read up to |desiredSize| bytes
+                from the port, putting the result in the [=byte sequence=]
+                |bytes|.
+              <li>
+                [=Queue a global task=] on the [=relevant global object=] of
+                [=this=] using the [=serial port task source=] to run the
+                following steps:
+                <ol>
+                  <li>
+                    If no errors were encountered run the following steps:
+                    <ol>
+                      <li>
+                        Let |buffer| be a [=new=] {{ArrayBuffer}} created from |bytes|.
+                      <li>
+                        Let |chunk| be a [=new=] {{Uint8Array}} view over |buffer|,
+                        who's length is the length of |bytes|.
+                      <li>
+                        Invoke [=ReadableStream/enqueue=] on
+                        [=this=].{{[[readable]]}} with |chunk|.
+                    </ol>
+                  <li>
+                    If a buffer overrun condition was encountered, invoke
+                    [=ReadableStream/error=] on [=this=].{{[[readable]]}} with a
+                    <dfn>`BufferOverrunError`</dfn> and invoke the steps to [=handle
+                    closing the readable stream=].
+                  <li>
+                    If a break condition was encountered, invoke
+                    [=ReadableStream/error=] on [=this=].{{[[readable]]}} with a
+                    <dfn>`BreakError`</dfn> and invoke the steps to [=handle closing
+                    the readable stream=].
+                  <li>
+                    If a framing error was encountered, invoke
+                    [=ReadableStream/error=] on [=this=].{{[[readable]]}} with a
+                    <dfn>`FramingError`</dfn> and invoke the steps to [=handle
+                    closing the readable stream=].
+                  <li>
+                    If a parity error was encountered, invoke
+                    [=ReadableStream/error=] on [=this=].{{[[readable]]}} with a
+                    <dfn>`ParityError`</dfn> and invoke the steps to [=handle
+                    closing the readable stream=].
+                  <li>
+                    If an operating system error was encountered, invoke
+                    [=ReadableStream/error=] on [=this=].{{[[readable]]}} with a
+                    <dfn>`UnknownError`</dfn> and invoke the steps to [=handle
```

Comment:
This one already exists, so can be cross-linked normally: `"{{UnknownError}}" {{DOMException}}`